server/src/main/java/org/elasticsearch/transport/ConnectionManager.java

```diff
@@ -166,44 +155,28 @@ public void connectToNode(DiscoveryNode node, ConnectionProfile connectionProfil
                                     connectionListener.onNodeDisconnected(node);
                                 }));
                             }
-                            if (conn.isClosed()) {
-                                throw new NodeNotConnectedException(node, "connection concurrently closed");
-                            }
-                            success = true;
-                            listeners = connectingNodes.remove(node);
                         }
-                    } catch (ConnectTransportException e) {
-                        throw e;
-                    } catch (Exception e) {
-                        throw new ConnectTransportException(node, "general node connection failure", e);
```

Comment:
AFAICS you removed the logic to wrap generic exceptions here into a ConnectTransportException? This might have an effect on other places in the code where we match on exception type for retry behavior.

Reply:
I mean this exception handling is mostly an artifact of the blocking version. The only line that theoretically throws an exception is:

```
connectionListener.onNodeConnected(node);
```

This interface is not actually implemented by production code (only tests). And if it is throwing an unchecked exception that is probably an issue.

Reply:
ok, makes sense